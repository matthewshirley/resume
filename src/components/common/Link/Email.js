import React from 'react';
import PropTypes from 'prop-types';

import Text from '../Text';

const EmailLink = ({ emailAddress }) => {
  if (!emailAddress) return (null);

  return (
    <Text as="a" href={`mailto:${emailAddress}`} m={0}>{emailAddress}</Text>
  );
};

EmailLink.propTypes = {
  emailAddress: PropTypes.string,
};

EmailLink.defaultProps = {
  emailAddress: null,
};

export default EmailLink;
