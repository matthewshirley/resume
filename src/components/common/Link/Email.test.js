import React from 'react';
import renderer from 'react-test-renderer';

import Email from './Email';

describe('Email Link', () => {
  it('renders correctly with an email address', () => {
    const tree = renderer
      .create(<Email emailAddress='test@user.com' />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('does not render without an email', () => {
    const tree = renderer
      .create(<Email emailAddress={null} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
