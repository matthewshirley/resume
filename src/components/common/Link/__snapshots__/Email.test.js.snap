// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email Link does not render without an email 1`] = `null`;

exports[`Email Link renders correctly with an email address 1`] = `
<a
  className="sc-AxjAm bnsefY"
  fontSize={1}
  href="mailto:test@user.com"
>
  test@user.com
</a>
`;
